src/main/scala/abalone/Pos.scala

```diff
@@ -190,8 +190,36 @@ case class Pos private (index: Int) extends AnyVal {
   def right: Option[Pos]     = Pos.at(file.index + 1, rank.index)
   def downRight: Option[Pos] = Pos.at(file.index, rank.index - 1)
   def upRight: Option[Pos]   = Pos.at(file.index + 1, rank.index + 1)
+
   def neighbours: List[Option[Pos]] = List(left, downLeft, upLeft, right, downRight, upRight)
 
+  // NOTE - *neighbourhood
+  // these below only work for neighbour pos but that's probably fine as in Abalone we only move to (potentially extended) neighbourhood
+  def dir(dir: Option[String]): Option[Pos]  =
+    dir match {
+      case Some("left") => this.left
+      case Some("downLeft") => this.downLeft
+      case Some("upLeft") => this.upLeft
+      case Some("right") => this.right
+      case Some("downRight") => this.downRight
+      case Some("upRight") => this.upRight
+      case _ => None
+    }
```

Comment:
I wonder if you can reuse the idea of `Direction` which is in the chess game logic, instead of having strings for each direction?

Reply:
actually in abalone package.scala we have the same :

```
  type Direction  = Pos => Option[Pos]
  type Directions = List[Direction]
```

Do you want I start having a look at how to use this properly then ?

Reply:
sure, using a type with the 6 options instead of a string, is safer (typesafety).